### DISCUSSION TITLE
Weird behaviour after exiting resize mode

### DISCUSSION THREAD

**Discussion by rwijtvliet** [mrjones2014/smart-splits.nvim#190] [General]
Entering resize works; resizing works; but when I exit the resize mode (by pressing escape), the editor resumes the last search (left bracket in the example below), and I seem to be unable to exit this:

<img width="535" alt="image" src="https://github.com/mrjones2014/smart-splits.nvim/assets/4106013/686fa14f-4bc4-448f-9cdf-d82e67a2e1a1">
 
`Esc` doesn't exit this mode, and it also persists when I open a new file or move to another split.

I have no idea how it's happened - maybe someone here does? 

My config:
```lua
  {
    'mrjones2014/smart-splits.nvim',
    lazy = true,
    dependencies = { 'kwkarlwang/bufresize.nvim' },
    opts = {
      ignored_filetypes = { 'nofile', 'quickfix', 'qf', 'prompt' },
      ignored_buftypes = { 'nofile' },
      resize_mode = {
        quit_key = '<ESC>',
        resize_keys = { 'h', 'n', 't', 's' },
        silent = true,
        hooks = {
          on_enter = function()
            vim.notify 'Entering window resize mode, press <esc> to exit.'
          end,
          on_leave = function()
            vim.notify 'Resize finished.'
            require('bufresize').register()
          end,
        },
        at_edge = 'stop',
      },
    },
  },
```

And my keymap:
```lua
vim.keymap.set('n', '<leader>wr', function()
  require('smart-splits').start_resize_mode()
end, { desc = 'Resize' })
```

My entire config can be found [here](https://github.com/rwijtvliet/dotfiles/tree/main/neovim/nvim).

---

**EDIT**

Looking into this a bit more, I suspect this to be related:

```lua
-- Set highlight on search, but clear on pressing <Esc> in normal mode
vim.opt.hlsearch = true
vim.keymap.set('', '<C-e>', '<Nop>')
vim.keymap.set('n', '<Esc>', '<cmd>nohlsearch<CR>')
```

**Comment by mrjones2014** [comment]
Hmm, I also have the same `nohlsearch` keybind and I don't have this issue :thinking:   https://github.com/mrjones2014/dotfiles/blob/d051dd3df7d1e7a1568de611aa3da944a93de142/nvim/lua/my/legendary/keymap.lua#L11-L12

Can you reproduce without `bufresize`? That's the main difference from my own config that I'm seeing (I don't use that plugin)

**Reply by rwijtvliet** [reply]
Thanks for the quick response! Yes, it also occurs with `on_leave = nil`

I could try to disable all other plugins, but it'd have to look into how to do that.

**Reply by mrjones2014** [reply]
Try a minimal init.lua like this https://github.com/mrjones2014/smart-splits.nvim/discussions/72

**Reply by rwijtvliet** [reply]
I used that file, added one line (`vim.keymap.set('n', '<C-b>', require('smart-splits').start_resize_mode)`). The problem does no longer persist in this situation.

**Reply by rwijtvliet** [reply]
I think the original problem is caused by the following: when exiting resize mode, the key map that I had before entering resize mode is not correctly restored. For example, I have `s` mapped to `l` (i.e., move right), but after exiting resize mode, this is "substitute". I have `n` mapped to `j` (i.e., move down), but after exiting resize mode this is "find next". This is probably what caused my initial problem: I press `n` to move to the row below, but vim unexpectedly tries to "find next".

**Reply by rwijtvliet** [reply]
Some more information: 

* apparently, it's only the navigational keys (set as `resize_keys = { 'h', 'n', 't', 's' },` in my config - note the not-standard bindings) that don't regain their normal-mode mapping as set in `keymap.lua`. 
* changing the `quit_key` to another key does not solve the issue.

**Reply by mrjones2014** [reply]
Ah yeah, it just unmaps the keys on exit. I will have to look for a workaround for this.

**Reply by mrjones2014** [reply]
Could you open an issue for this?

**Reply by rwijtvliet** [reply]
Of course - here you go: https://github.com/mrjones2014/smart-splits.nvim/issues/192

**Comment by rwijtvliet** [comment]
Moved discussion to bug-report https://github.com/mrjones2014/smart-splits.nvim/issues/192
